bundlewrap/items/dconf.py

```diff
@@ -0,0 +1,135 @@
+
+    @property
+    def path(self):
+        return '/' + self.name.split('/', 1)[1]
+
+    def _parse_result(self, result):
+        stdout = result.stdout.decode('UTF-8').strip()
+        if len(stdout) == 0:
+            return None
+        elif stdout.isdigit():
+            return int(stdout)
+        else:
+            try:
+                return literal_eval(stdout)
+            except Exception:  # skipcq: PYL-W0703
+                return stdout
+
+    def run(self, command, **kwargs):
+        result = self.node.run(
+            f'DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/$(id -u {self.user})/bus {command}',
+            user=self.user,
+            **kwargs,
+        )
+        self._command_results.append({
+            'command': command,
+            'result': result,
+        })
+        return result
+
+    def get_auto_attrs(self, items):
+        needs = set()
+        after = set()
+        for item in items:
+            # We need to depend on the user item and the GNOME packages.
+            # To increase the chance of the setting being available, we
+            # try to run this item after all package installations have
+            # succeeded.
+            if item.ITEM_TYPE_NAME == 'user' and item.name == self.user:
+                needs.add(item.id)
+            elif item.ITEM_TYPE_NAME.startswith('pkg_') and item.name.lower().startswith('gnome'):
+                needs.add(item.id)
+            elif item.ITEM_TYPE_NAME.startswith('pkg_'):
+                after.add(item.id)
+        return {
+            'after': after,
+            'needs': needs,
+        }
+
+    def cdict(self):
+        if self.attributes['reset']:
+            return None
+
+        return {
+            'value': self.attributes['value'],
+        }
+
+    def sdict(self):
+        result = self.run(f'dconf read {self.path}', may_fail=True)
+        value = self._parse_result(result)
+
+        if value is None:
+            return None
+
+        return {
+            'value': value,
+        }
+
+    def fix(self, status):
+        if status.must_be_created or status.keys_to_fix:
+            self.run("dconf write {path} '{value}'".format(
+                path=self.path,
+                value=dumps(self.attributes['value']),
+            ))
+        elif status.must_be_deleted:
+            self.run('dconf reset {self.path}')
+
+    def patch_attributes(self, attributes):
+        if isinstance(attributes.get('value', []), set):
+            attributes['value'] = sorted(attributes['value'])
+        return attributes
+
+    @classmethod
+    def validate_attributes(cls, bundle, item_id, attributes):
+        if not attributes.get('value') and not attributes.get('reset'):
+            raise BundleError(_(
+                f'Item {item_id} in bundle {bundle} has no value set. '
+                'Please explicitely set the "reset" attribute if you '
+                'wish to reset this setting to the default value.'
+            ))
```

Comment:
You can't combine f-strings with `_()` like that. The values have to be inserted into the string *after* it has been through `_()`, so you need to use `_("").format()`.

Reply:
I'll fix after merge